go/plugins/compat_oai/openai/openai.go

```diff
@@ -136,3 +148,55 @@ func DefineModel(g *genkit.Genkit, name string, info ai.ModelInfo) (ai.Model, er
 		return resp, nil
 	}), nil
 }
+
+// DefineEmbedder defines an embedder with a given name.
+func DefineEmbedder(g *genkit.Genkit, name string) (ai.Embedder, error) {
+	if !state.initted {
+		panic("openai.Init not called")
+	}
+
+	if !slices.Contains(knownEmbedders, name) {
+		return nil, fmt.Errorf("unsupported embedder: %s", name)
+	}
```

Comment:
Suggestion: we should remove this, since we are going to make it generic, anyone can define an embedder